boards/arm64/intel_socfpga_agilex_socdk/intel_socfpga_agilex_socdk.dts

```diff
@@ -27,3 +27,10 @@
 	status = "okay";
 	current-speed = <115200>;
 };
+
+&qspi {
+	status = "okay";
+	mt25qu02g {
+		status = "okay";
```

Comment:
Please flush this node out similar to have a compatible, size, and reg

Reply:
do you mean adding size and reg under mt25qu02g node?

Reply:
Yes, should be something like:

```
mt25qu02g@0 {
   compatible = "micron,mt25qu02g";
   reg = <0>;
   size = <XYZ>;
};
```

NOTE: size is in bits, so do something like `size = <DT_SIZE_M(XYZ*8)>;       /* in bits */`

Reply:
I see, but my size, reg has already defined in the dtsi. Which is the range of this address range

qspi: qspi@ff8d2000 {
		compatible = "cdns,qspi-nor";
		reg = <0xff8d2000 0x100>,
		      <0xff900000 0x100000>;

Reply:
Hi @galak , I'm not sure what are the reg and size for under the node, mt25quo02g since i already have the reg and size defined at the qspi node, please advice.

Reply:
Hi @galak, can you please advice?

Reply:
reg should most likely be 0.  You need to look at the datasheet for the exact flash device to see what size it is.

Reply:
is the size here mean the exact size of the flash? for example this mt25qu02g, this can store 2GB of data. that case i have to DT_SIZE_G<2 * 8>?

Reply:
assuming the device is something like:

https://www.digikey.com/en/products/detail/micron-technology-inc/MT25QU02GCBB8E12-0AUT/10321185

Than I'd say it should be:

`size = <DT_SIZE_M(256*8)>; /* in bits */`

Reply:
Hi @galak 

I hit into error when adding the following dts setting.

```
&qspi {
        status = "okay";
        mt25qu02g: mt25qu02g@0 {
                compatible = "micron,mt25qu02g";
                reg = <12>;
                size = <2048>; /* in bits */
                status = "okay";
        };

```

-- Found BOARD.dts: /p/psg/swip/w/boonkhai/git/boonkhai_github/zephyr-from-mainline-v3/zephyrproject/zephyr/boards/arm64/intel_socfpga_agilex_socdk/intel_socfpga_agilex_socdk.dts
devicetree error: 'reg' property in <Node /qspi@ff8d2000/mt25qu02g@0 in '/p/psg/swip/w/boonkhai/git/boonkhai_github/zephyr-from-mainline-v3/zephyrproject/zephyr/misc/empty_file.c'> has length 4, which is not evenly divisible by 12 (= 4*(<#address-cells> (= 2) + <#size-cells> (= 1))). Note that #*-cells properties come either from the parent node or from the controller (in the case of 'interrupts').
-- In: /

Reply:
I play around with the reg number but no luck, i tried with 0, 4, 8, 12  not sure why it complaint about the 'reg' property not evenly divisible by 12. Please advice

Reply:
Resolved Thanks.